SWIPs/swip-data-types-descriptions.md

```diff
@@ -0,0 +1,70 @@
+---
+SWIP: <To be assigned>
+title: Swarm node implementer spec - 
+author: Louis Holbrook @nolash <dev@holbrook.no>
+status: Draft
+type: Track Specs
+created: 2019-08-02
+---
+
+## Abstract
+
+This SWIP is a part of a general specification of a Swarm node. The SWIP system is used for convenience only.
+
+## Motivation
+
+Enable other implementations of the swarm node.
+
+## Backwards Compatibility
+
+N/A
+
+## Test Cases
+
+N/A
+
+## Implementation
+
+N/A
+
+## Copyright
+
+Copyright and related rights waived via CC0
+
+## Specification
+
+### Data representations
+
+This document uses the `ABNF` convention for defining data types, as
+described in the IETF RFC 5234. 
+
+#### Primitives
+
+The basic identifiers are defined as follows:
+
+| id  | def |
+| --- | :--- |
+|    UINT64MAX  | 18446744073709551615 |
+|    UINT32MAX  | 4294967295 |
+|    UINT16MAX  | 65535 |
+|    UINT8MAX   | 255 |
+|    UINT64     | %d0-UINT64MAX |
```

Comment:
not sure, but perhaps it is good to elaborate in this since it won't be obvious to everone reading spec what "%d0" is